src/main/java/com/github/sdonkov/PrefixTree.java

```diff
@@ -3,16 +3,50 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.AbstractCollection;
 import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 import java.util.Optional;
 
-public class PrefixTree {
+public class PrefixTree extends AbstractCollection<String> {
 
     private final static Logger LOGGER = LogManager.getLogger(PrefixTree.class);
     private final Node root;
+    private int size;
+    private final ArrayList<String> data;
```

Comment:
This is no longer a prefix tree now that way, is it?
One of the key points about this task is to figure out how to iterate over the keys, i.e. traverse the trie.